scd30/scd30.h

```diff
@@ -234,6 +234,27 @@ int16_t scd30_enable_automatic_self_calibration(uint8_t enable_asc);
  */
 int16_t scd30_set_forced_recalibration(uint16_t co2_ppm);
 
+/**
+ * Read out the serial number
+ *
+ * @param serial    the address for the result of the serial number.
+ *                  --------------------------------------
+ *                  THE BUFFER MUST HOLD AT LEAST 33 BYTES
+ *                  --------------------------------------
+ *                  Usage example:
+ *                  ```
+ *                  char scd30_serial[33];
+ *                  if (scd30_read_serial(scd30_serial) == 0) {
+ *                      printf("SCD30 serial: %s\n", scd30_serial);
+ *                  } else {
+ *                      printf("Error reading SCD30 serial\n");
+ *                  }
+ *                  ```
+ *                  Contains a zero-terminated string.
+ * @return          0 if the command was successful, else an error code.
+ */
+int16_t scd30_read_serial(char *serial);
```

Comment:
I'm thinking if it would help to use a pointer to an 33 byte buffer here. So `char (*serial)[33]`. This would then need to be called with `scd30_read_serial(&scd30_serial)` and would only work if the buffer is exactly 33 bytes.

Reply:
I thought about it, but it's always a pain to use... what I also thought was to pass a buffer length param, so you can read as many bytes as you want.. e.g.

```c
char mine_is_too_small[20];
scd30_read_serial(mine_is_too_small, sizeof(mine_is_too_small));
assert(mine_is_too_small[sizeof(mine_is_too_small) - 1] == '\0');
```

Reply:
I suppose you meant `int16_t scd30_read_serial(char serial[33]);` as signature

Reply:
> I suppose you meant int16_t scd30_read_serial(char serial[33]); as signature

No, that would [decay](https://stackoverflow.com/questions/1461432/what-is-array-decaying) to the same as `char* serial`, thus loosing the size information. With `char (*serial)[33]` you keep the size information in the type.

Reply:
But let's keep it as it is.